pkg/core/service.go

```diff
@@ -86,6 +93,105 @@ func newService(options CreateServiceOptions) v1alpha1.Service {
 			},
 		},
 	}
+
+	if len(options.Env) > 0 {
```

Comment:
Why special case the empty case since this code isn't performance sensitive? The rest of the code should be able to cope smoothly with that case.